### PR TITLE
Closes #62 | Create datalaoder for SEA MADLAD

### LINKED ISSUE #62: Create dataset loader for SEA MADLAD
Dataloader name: `sea_madlad/sea_madlad.py`
DataCatalogue: http://seacrowd.github.io/seacrowd-catalogue/card.html?sea_madlad

|     Dataset| sea_madlad |
|-------------|---|
| Description | SEA MADLAD is a subset of MADLAD-400 (Multilingual Audited Dataset: Low-resource And Document-level), which is a document-level multilingual dataset based on Common Crawl. SEA MADLAD only filters the language of the "clean" subset, which covers 36 languages indigenous to SEA from 419 languages in total.   Please note some of SEA lang codes weren't available in this version because those belongs to the languages whose decision was to "remove from its clean version" based on MADLAD auditing process.  MADLAD uses all snapshots of CommonCrawl available as of August 1, 2022. The primary advantage of this dataset over similar datasets is that it is more multilingual, it is audited and more highly filtered, and it is document-level. The main disadvantage is also its strength -- being more filtered, it may lack the recall needed for some applications. |
| Subsets | ace, akb, ban, bbc, bew, bik, btx, ceb, fil, gor, hil, iba, ilo, ind, jav, kac, khm, kxd, lao, mad, mak, meo, min, mkn, msa, msi, mya, nij, nut, pag, shn, sun, tet, tha, vie, war |
| Languages | ace, akb, ban, bbc, bew, btx, ceb, fil, gor, hil, iba, ilo, ind, jav, kac, khm, kxd, lao, mad, mak, meo, min, mkn, msi, mya, nij, nut, pag, shn, sun, tet, tha, vie, war |
| Tasks | Language Modeling, Language Identification |
| License | Creative Commons Attribution 4.0 (cc-by-4.0) |
| Homepage | https://huggingface.co/datasets/allenai/MADLAD-400 |
| HF URL | https://huggingface.co/datasets/allenai/MADLAD-400 |
| Paper URL | https://arxiv.org/abs/2309.04662 |

### PR DESCRIPTION
**Pull request by sabilmakbar**
Closes #62.

### Checkbox
- [x] Confirm that this PR is linked to the dataset issue.
- [x] Create the dataloader script `seacrowd/sea_datasets/my_dataset/my_dataset.py` (please use only lowercase and underscore for dataset naming).
- [x] Provide values for the `_CITATION`, `_DATASETNAME`, `_DESCRIPTION`, `_HOMEPAGE`, `_LICENSE`, `_URLs`, `_SUPPORTED_TASKS`, `_SOURCE_VERSION`, and `_SEACROWD_VERSION` variables.
- [x] Implement `_info()`, `_split_generators()` and `_generate_examples()` in dataloader script.
- [x] Make sure that the `BUILDER_CONFIGS` class attribute is a list with at least one `SEACrowdConfig` for the source schema and one for a seacrowd schema.
- [x] Confirm dataloader script works with `datasets.load_dataset` function.
- [x] Confirm that your dataloader script passes the test suite run with `python -m tests.test_seacrowd seacrowd/sea_datasets/<my_dataset>/<my_dataset>.py`.
- [] If my dataset is local, I have provided an output of the unit-tests in the PR (please copy paste). This is OPTIONAL for public datasets, as we can test these without access to the data files.

## Note
The estimated total download size will amounting to ~200 GB if all languages available are being downloaded. See this [section](https://huggingface.co/datasets/allenai/MADLAD-400#final-dataset-information) for estimated total size per lang (on column `clean`)

### CODE REVIEW

File: seacrowd/sea_datasets/sea_madlad/sea_madlad.py [hunk 1]
```diff
@@ -0,0 +1,256 @@
...
+from datasets import load_dataset
+from datasets.download.download_manager import DownloadManager
+
+from seacrowd.utils import schemas
+from seacrowd.utils.configs import SEACrowdConfig
+from seacrowd.utils.constants import TASK_TO_SCHEMA, Licenses, Tasks
+
+_CITATION = r"""
+@misc{kudugunta2023madlad400,
+      title={MADLAD-400: A Multilingual And Document-Level Large Audited Dataset},
```
Comment (ljvmiranda921) [inline seacrowd/sea_datasets/sea_madlad/sea_madlad.py:19]: Trailing whitespace

```suggestion
      title={MADLAD-400: A Multilingual And Document-Level Large Audited Dataset},
```

File: seacrowd/sea_datasets/sea_madlad/sea_madlad.py [hunk 2]
```diff
@@ -0,0 +1,256 @@
...
+
+CONFIG_SUFFIXES_FOR_TASK = [TASK_TO_SCHEMA.get(task).lower() for task in _SUPPORTED_TASKS]
+
+
+def conform_init_config():
+    """Assertion Function for Instantiated Configs"""
+    if len(_LANGUAGES) == 0:
+        raise AssertionError("No Languages detected from config!")
+    if len(CONFIG_SUFFIXES_FOR_TASK) != len(_SUPPORTED_TASKS):
+        raise AssertionError("Config prefixes doesn't matched in terms of `len` with `_SUPPORTED_TASKS`!")
```
Comment (ljvmiranda921) [inline seacrowd/sea_datasets/sea_madlad/sea_madlad.py:140]:
```suggestion
        raise AssertionError("Config prefixes don't matched in terms of `len` with `_SUPPORTED_TASKS`!")
```

**Review by ljvmiranda921 — APPROVED** [review]
Thanks for the nice PR @sabilmakbar ! I ran the dataset loader for every language in SEA MADLAD using the script you mentioned, and it worked with no errors for both source and seacrowd schemas. I can't run the standard test because the dataset size is too big for my machine.

I have some small nits based on formatting, but aside from that LGTM :)

**Comment by sabilmakbar** [comment]
comments addressed, kindly recheck @ljvmiranda921!

**Review by ljvmiranda921 — APPROVED** [review]
LGTM!

**Review by holylovenia — APPROVED** [review]
Tried and tested. It works well!! Thank you so much, @sabilmakbar. Merging this now.

**Pull request merged**